src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,575 @@
+                // Similarly the individual node (elements in the list) for the reviews, authors ListValueNode(s) are also interpreted as ObjectValueNode(s).
+                ValidateObjectFieldNodes(
+                    schemaObject: ExecutionHelper.InputObjectTypeFromIInputField(schema),
+                    entityName: entityName,
+                    context: context,
+                    objectFieldNodes: objectValueNode.Fields,
+                    runtimeConfig: runtimeConfig,
+                    columnsDerivedFromSourceEntity: columnsDerivedFromParentEntity,
+                    columnsToDerive: columnsToBeDerivedFromEntity,
+                    nestingLevel: nestingLevel + 1,
+                    parentEntityName: parentEntityName,
+                    sqlMetadataProviderFactory: sqlMetadataProviderFactory);
+            }
+            else if (parameters is ListValueNode listValueNode)
+            {
+                // For the example createbook mutation written above, the list values for reviews and authors fields are interpreted as ListValueNode.
+                // All the nodes in the ListValueNode are parsed one by one.
+                listValueNode.GetNodes().ToList().ForEach(objectValueNodeInListValueNode => ValidateGraphQLValueNode(
+                    schema: schema,
+                    entityName: entityName,
+                    context: context,
+                    parameters: objectValueNodeInListValueNode,
+                    runtimeConfig: runtimeConfig,
+                    columnsDerivedFromParentEntity: columnsDerivedFromParentEntity,
+                    columnsToBeDerivedFromEntity: columnsToBeDerivedFromEntity,
+                    nestingLevel: nestingLevel,
+                    parentEntityName: parentEntityName,
+                    sqlMetadataProviderFactory: sqlMetadataProviderFactory));
+            }
+        }
+
+        /// <summary>
+        /// Helper method to iterate over all the fields present in the input for the current field and
+        /// validate that the presence/absence of fields make logical sense for the multiple-create request.
+        /// </summary>
+        /// <param name="schemaObject">Input object type for the field.</param>
+        /// <param name="entityName">Current entity's name.</param>
+        /// <param name="context">Middleware Context.</param>
+        /// <param name="objectFieldNodes">List of ObjectFieldNodes for the the input field.</param>
+        /// <param name="runtimeConfig">Runtime config</param>
+        /// <param name="columnsDerivedFromSourceEntity">Set of columns in this entity whose values should be
+        /// derived from insertion in the source entity (i.e. source entity would have been the referenced entity).</param>
+        /// <param name="columnsToDerive">Set of columns in this entity whose values are to be
+        /// derived from insertion in the this entity or its subsequent referenced entities and returned to the source entity
```

Comment:
```suggestion
        /// <param name="columnsToDerive">Set of columns in this entity whose values are to be
        /// derived from insertion in its referenced entities and returned to the source entity
```

Reply:
The current summary I think better explains what I am trying to convey here. Please let me know if you have any questions/confusions regarding this.